src/rgw/rgw_swift_auth.cc

```diff
@@ -82,7 +82,7 @@ void TempURLEngine::get_owner_info(const DoutPrefixProvider* dpp, const req_stat
   const string& bucket_name = s->init_state.url_bucket;
 
   /* TempURL requires that bucket and object names are specified. */
-  if (bucket_name.empty() || s->object->empty()) {
+  if (bucket_name.empty() || !s->object || s->object->empty()) {
```

Comment:
`rgw::sal::Object::empty(s->object)` seems to be the canonical way to check for both nullness and emptiness

Reply:
that's a useful observation, but I find this logic clear--the big risk would be missing part of the logic that is being fixed